### PR TITLE
Import events improvements 

### PR DESCRIPTION
**Pull request by kjgarza**
- filters  import events from crossref.citations out
- check the Registration Agency of the DOIS to import from specific sources
- caches Member id and registrations agencies

### CODE REVIEW

File: app/models/base.rb [hunk 1]
```diff
@@ -45,6 +45,8 @@ def get_query_url(options={})
     options[:number] ||= 1
     options[:size] ||= 1000
     updated = "updated:[#{options[:from_date]}T00:00:00Z TO #{options[:until_date]}T23:59:59Z]"
+    options[:exclude_registration_agencies] ||= true
+    options[:resource_type_id] ||= ""
```
Comment (mfenner) [inline app/models/base.rb]: can you explain this line.

File: app/models/concerns/cacheable.rb [hunk 2]
```diff
@@ -31,5 +31,18 @@ def cached_ror_response(id)
...
     end
+
+    def cached_doi_ra(doi)
+      Rails.cache.fetch("ras/#{doi}", expires_in: 1.day) do
+        prefix = Base.validate_prefix(doi)
+        Base.get_doi_ra(prefix)
+      end
+    end
+  
+    def cached_crossref_member_id(doi)
```
Comment (mfenner) [inline app/models/concerns/cacheable.rb]: Crossref member ids are not DOIs.

File: app/models/related_identifier.rb [hunk 3]
```diff
@@ -55,6 +59,7 @@ def self.push_item(item)
     attributes = item.fetch("attributes", {})
     doi = attributes.fetch("doi", nil)
     return nil unless doi.present?
+    return nil unless cached_doi_ra(doi) == "DataCite"
```
Comment (mfenner) [inline app/models/related_identifier.rb]: put on single line with above line.

File: app/models/related_identifier.rb [hunk 4]
```diff
@@ -63,8 +68,8 @@ def self.push_item(item)
     push_items = Array.wrap(related_doi_identifiers).reduce([]) do |ssum, iitem|
       related_identifier = iitem.fetch("relatedIdentifier", nil).to_s.strip.downcase
       obj_id = normalize_doi(related_identifier)
-      prefix = validate_prefix(related_identifier)
-      registration_agencies[prefix] = get_doi_ra(prefix) unless registration_agencies[prefix]
+      # prefix = validate_prefix(related_identifier)
+      registration_agencies[prefix] = cached_doi_ra(related_identifier) unless registration_agencies[prefix]
```
Comment (mfenner) [inline app/models/related_identifier.rb]: Please lookup prefix not identifier, as caching is much more efficient.

File: lib/tasks/related_identifier.rake [hunk 5]
```diff
@@ -3,8 +3,9 @@ namespace :related_identifier do
   task :import_by_month => :environment do
     from_date = ENV['FROM_DATE'] || Date.current.beginning_of_month.strftime("%F")
     until_date = ENV['UNTIL_DATE'] || Date.current.end_of_month.strftime("%F")
+    resource_type_id = ENV['RESOURCE_TYPES'] || ""
```
Comment (mfenner) [inline lib/tasks/related_identifier.rake]: Why ""?

File: app/models/base.rb (hunk 1 above)
Comment (kjgarza) [reply app/models/base.rb]: without excluding other registrations agencies,  the method get_data (https://github.com/datacite/levriero/blob/1a7d39c4d04b5e4a21ec47b97f5cf970e6927440/app/models/base.rb#L128) will bring all updated dois, including those from crossref.citations which might have in their metadata related identifiers to other Crossref DOIs.  That would create xref-to-xref events.

**Pull request merged**